ConnectFourGame/src/connect/four/gui/MainMenuPanel.java

```diff
@@ -138,9 +158,29 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
 
 
     private void butPlayActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_butPlayActionPerformed
-		gui.setPlayer1Name(tfplayer1.getText().length()>0? tfplayer1.getText() : "Player 1");
-		gui.setPlayer2Name(tfplayer2.getText().length()>0? tfplayer2.getText() : "Player 2");
-		//gui.game.start();
+        if (tfplayer1.getText().length() >= MAX_CHARACTERS_IN_NAME )	// if more than max character input
+        	tfplayer1.setText(tfplayer1.getText().substring(0, MAX_CHARACTERS_IN_NAME));	// sets name to substring
+        if (tfplayer2.getText().length() >= MAX_CHARACTERS_IN_NAME )	// if more than max character input
+        	tfplayer2.setText(tfplayer2.getText().substring(0, MAX_CHARACTERS_IN_NAME));	// sets name to substring
+        if(tfplayer1.getText().equals(tfplayer2.getText()) && tfplayer1.getText().length() > 0) { // if names are same and not empty
+        	String tempName = tfplayer1.getText();
+            if (tempName.length() >= MAX_CHARACTERS_IN_NAME-2)	// if more than max character input - 2
+            	tempName = tempName.substring(0, MAX_CHARACTERS_IN_NAME-2);	// sets name to substring
+        	tfplayer1.setText(tempName + "(1)");
+        	tfplayer2.setText(tempName + "(2)");
+        }
+        // player names actually set below
+		gui.setPlayer1Name(tfplayer1.getText().length()>0? tfplayer1.getText() : "Player 1");	// if player 1 has empty name, set name to "Player 1"
+		gui.setPlayer2Name(tfplayer2.getText().length()>0? tfplayer2.getText() : "Player 2");	// if player 2 has empty name, set name to "Player 2"
+		// wanted to use a switch statement below, but forgot how and don't have internet
+		if(compDifficulty.getSelectedIndex() == 0)
```

Comment:
This section actually sets the difficulty. It uses the index of the current selection to determine difficulty, so don't change order or number of choices without editing this as well.